target/arc/semfunc.c

```diff
@@ -55,27 +55,52 @@ arc_gen_cmpl2_i64(TCGv_i64 ret, TCGv_i64 arg1,
 }
 
 void
+/**
+ * In a signed addition, there is an overflow when these conditions are met:
+ * (1) Both operands are of the same sign.
+ * (2) The result's sign is different from the operand's.
+ */
+arc_gen_add_signed_overflow_tl(TCGv overflow, TCGv result,
+                               TCGv op1, TCGv op2, uint8_t operand_size)
+{
+    TCGv diff_operand_sign = tcg_temp_new();
+    TCGv diff_result_sign = tcg_temp_new();
+
+    tcg_gen_xor_tl(diff_result_sign, op1, result);
+    tcg_gen_xor_tl(diff_operand_sign, op1, op2);
+    tcg_gen_andc_tl(overflow, diff_result_sign, diff_operand_sign);
+
+    tcg_gen_shri_tl(overflow, overflow, operand_size - 1);
+    tcg_gen_andi_tl(overflow, overflow, 1);
+
+    tcg_temp_free(diff_operand_sign);
+    tcg_temp_free(diff_result_sign);
+}
+
+void
+/**
+ * In a signed addition, there is an overflow when these conditions are met:
+ * (1) Both operands are of the same sign.
+ * (2) The result's sign is different from the operand's.
+ */
 arc_gen_add_signed_overflow_i64(TCGv_i64 overflow, TCGv_i64 result,
```

Comment:
please add a comment explaining why this `*_ia64` edition. I think it is only used for 64-bit vector operations both for 32-bit and 64-bit targets. right?

Reply:
For the time being yes, but if there are more 64 bit operation instructions in a 32 bit architecture it will be used there as well.